From b8178bb63cd48d6ba024f35f620435394f61fe8b Mon Sep 17 00:00:00 2001
From: Du Chenyang <chenyang.du@intel.com>
Date: Mon, 16 Jul 2012 18:44:51 +0800
Subject: [PORT FROM MAIN] Adding build configuration for boot.bin

BZ: 46708

THIS SHOULD BE REPLACED BY https://android-secure.intel.com:8080/#/c/32/

Adding an option to support intel boot.bin build, which will disable
the default build of boot.img if TARGET_MAKE_NO_DEFAULT_BOOTIMAGE is true.

[BOOT CAMP SPECIFIC]

Change-Id: If4a85f757ca8eecce30542b9fd54e7652d1bed3f
Orig-Change-Id: I6b2cf1ad4f63517686085fa95ea6948153e2acbc
Signed-off-by: Sun Yijin <yijin.sun@intel.com>
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Signed-off-by: Du Chenyang <chenyang.du@intel.com>
Reviewed-on: http://android.intel.com:8080/56543
Tested-by: Gao, Shuo <shuo.gao@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-on: http://android.intel.com:8080/74477
Reviewed-by: Yao, Yong <yong.yao@intel.com>
---
 core/Makefile |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index ede8fc5..8cc46a2 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -406,6 +406,7 @@ ifdef BOARD_KERNEL_PAGESIZE
   INTERNAL_BOOTIMAGE_ARGS += --pagesize $(BOARD_KERNEL_PAGESIZE)
 endif
 
+ifneq ($(TARGET_MAKE_NO_DEFAULT_BOOTIMAGE),true)
 INSTALLED_BOOTIMAGE_TARGET := $(PRODUCT_OUT)/boot.img
 
 ifeq ($(TARGET_BOOTIMAGE_USE_EXT2),true)
@@ -424,6 +425,12 @@ $(INSTALLED_BOOTIMAGE_TARGET): $(MKBOOTIMG) $(INTERNAL_BOOTIMAGE_FILES)
 	$(hide) $(call assert-max-image-size,$@,$(BOARD_BOOTIMAGE_PARTITION_SIZE),raw)
 endif # TARGET_BOOTIMAGE_USE_EXT2
 
+else # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
+$(INSTALLED_BOOTIMAGE_TARGET): $(MAKE_NO_DEFAULT_BOOTIMAGE_ITEMS)
+	$(MAKE_NO_DEFAULT_BOOTIMAGE)
+endif # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
 else	# TARGET_NO_KERNEL
 # HACK: The top-level targets depend on the bootimage.  Not all targets
 # can produce a bootimage, though, and emulator targets need the ramdisk
-- 
1.7.4.1

